### PR TITLE
PRFC: Moving `sample-data` into `create-app` templates

### PR DESCRIPTION
**Pull request by benjdlambert**
A reference implementation for #11062 

- [x] Single entry point for extended sample data reading from catalog-model
- [ ] Simple mock data with 1 of each kind, and inline documentation.
- [ ] 1 guest user + 1 group that own some things to see the ownership model
- [x] Template should be simple nodejs backend (catalog-info.yaml, package.json, index.js)

### CODE REVIEW

**Review by jhaals — APPROVED** [review]
LGTM 👍

File: packages/catalog-model/examples/all.yaml [hunk 1]
```diff
@@ -0,0 +1,12 @@
+apiVersion: backstage.io/v1alpha1
+kind: Location
+metadata:
+  name: example-apis
+  description: A collection of all Backstage example APIs
```
Comment (freben) [inline packages/catalog-model/examples/all.yaml:5]:
```suggestion
  description: A collection of all Backstage example entities, except users and groups
```

File: app-config.yaml [hunk 2]
```diff
@@ -253,33 +251,25 @@ catalog:
...
     # Backstage example templates
     - type: file
       target: ../../plugins/scaffolder-backend/sample-templates/all-templates.yaml
     # Backstage example groups and users
     - type: file
       target: ../catalog-model/examples/acme-corp.yaml
+      rules:
+      - allow:
+          - User
+          - Group
```
Comment (freben) [inline app-config.yaml:272]: maybe move this entire block up to the other one above so both catalog-model ones are next to each other

File: app-config.yaml [hunk 3]
```diff
@@ -253,33 +251,25 @@ catalog:
     # File locations are relative to the current working directory of the
     # backend, for example packages/backend/.
 
-    # Backstage example components
+    # Backstage example entities
     - type: file
-      target: ../catalog-model/examples/all-components.yaml
+      target: ../catalog-model/examples/all.yaml
```
Comment (freben) [inline app-config.yaml:256]: it's mildly confusing that `all` doesn't have all the data i guess :)

File: packages/create-app/templates/default-app/examples/entities.yaml [hunk 4]
```diff
@@ -0,0 +1,41 @@
+---
+apiVersion: backstage.io/v1alpha1
+# https://backstage.io/docs/features/software-catalog/descriptor-format#kind-system
```
Comment (freben) [inline packages/create-app/templates/default-app/examples/entities.yaml:3]: i think it'd be nicer to move all of these comments up one row

File: packages/create-app/templates/default-app/app-config.yaml.hbs [hunk 5]
```diff
@@ -90,36 +90,26 @@ catalog:
...
-      target: https://github.com/backstage/software-templates/blob/main/scaffolder-templates/springboot-grpc-template/template.yaml
-      rules:
-        - allow: [Template]
-    - type: url
-      target: https://github.com/backstage/software-templates/blob/main/scaffolder-templates/docs-template/template.yaml
-      rules:
-        - allow: [Template]
+    ## Uncomment these lines to add an example org
+    # - type: url
+    #   target: https://github.com/backstage/backstage/blob/master/packages/catalog-model/examples/acme-corp.yaml
```
Comment (freben) [inline packages/create-app/templates/default-app/app-config.yaml.hbs:115]: this one should also have an `allow` block

File: packages/create-app/templates/default-app/examples/org.yaml [hunk 6]
```diff
@@ -0,0 +1,17 @@
...
+spec:
+  memberOf: [guests]
+---
+apiVersion: backstage.io/v1alpha1
+# https://backstage.io/docs/features/software-catalog/descriptor-format#kind-group
+kind: Group
+metadata:
+  name: guests
+spec:
+  type: squad
```
Comment (freben) [inline packages/create-app/templates/default-app/examples/org.yaml:16]: maybe team is more established i guess

**Review by freben — APPROVED** [review]

**Pull request merged**